azure-pipelines.yml

```diff
@@ -0,0 +1,174 @@
+variables:
+- name: clone_depth
+  value: 10 
+  
+
+# A pipeline with no CI trigger
+trigger: none
+
+# specific path
+pr:
+  branches:
+    include:
+    - master
+    - releases/*
+  paths:
+    exclude:
+    - CHANGELOG.md
+    - CONTRIBUTING.md
+    - .gitattributes
+    - .gitattributes
+    - LICENSE
+    - VERSION
+  drafts: false
+
+stages:
+  - stage: nimi_bot_windows
+    pool:
+      name: nimi-bot
+      demands:
+        - agent.os -equals Windows_NT
+    jobs:
+    - job: ensure_code_gen_up_to_date
+      workspace:
+        clean: all
+      steps:
+        - checkout: self  
+          clean: true
+          fetchDepth: $(clone_depth)
+        - task: PythonScript@0
+          displayName: Ensure codegen up to date
+          inputs:
+            scriptSource: inline
+            script: |
+              import os
+              import sys
+              import subprocess
+
+              # To verify code is generated
+              os.system('git config user.email "dummy@ni.com"')
+              os.system('git config user.name "dummy"')
+              commit_id_before_code_generator = os.system("git log -1 --format=%h")
+              os.system("tox -e codegen")
+              os.system('git commit -a -m "test"')
+              commit_id_after_code_generator = os.system("git log -1 --format=%h")
+              if commit_id_before_code_generator != commit_id_after_code_generator:
+                print("Run codegen and include the generated files in the PR", file=sys.stderr)
+                sys.exit(-1)
+
+              # To verify code generated is not added in .gitignore list
```

Comment:
You can get rid of this, it doesn't do what I was thinking of, which is check if the PR is creating new unintended git-ignored files. I don't think there's a good way to check that.

Reply:
done the changes